.github/workflows/update-vendor-hash.yml

```diff
@@ -0,0 +1,30 @@
+name: Update vendorHash
+on: pull_request
+
+permissions:
+  contents: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+      - name: Install Nix
+        uses: cachix/install-nix-action@v22
+        with:
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Update checksum
+        run: |
+          ./scripts/update-vendor-hash
```

Comment:
I tested the script locally but not yet the action (also I use the same actions in other places). It can be tested in existing dependabot PRs by using @dependabot recreate.